openc3-cosmos-cmd-tlm-api/app/channels/application_cable/connection.rb

```diff
@@ -12,25 +12,36 @@
 # but WITHOUT ANY WARRANTY; without even the implied warranty of
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU Affero General Public License for more details.
-
-# Modified by OpenC3, Inc.
-# All changes Copyright 2022, OpenC3, Inc.
-# All Rights Reserved
 #
-# This file may also be used under the terms of a commercial license 
+# This file may also be used under the terms of a commercial license
 # if purchased from OpenC3, Inc.
 
+require 'openc3/utilities/authorization'
+
 module ApplicationCable
   class Connection < ActionCable::Connection::Base
-    identified_by :uuid # , :token
+    include OpenC3::Authorization
+
+    identified_by :uuid
+    identified_by :scope
 
     def connect
-      # We don't have user accounts so use a random UUID
+      authorization('system')
       self.uuid = SecureRandom.uuid
-      # TODO: token? I saw some rails code using this
-      # self.token = request.params[:token] ||
-      #   request.cookies["token"] ||
-      #   request.headers["X-API-TOKEN"]
+      self.scope = request.headers['HTTP_X_OPENC3_SCOPE'] || request.query_parameters[:scope]
+    end
+
+    def authorization(permission)
+      begin
+        authorize(
+          permission: permission,
+          scope: request.headers['HTTP_X_OPENC3_SCOPE'] || request.query_parameters[:scope],
+          token: request.headers['HTTP_AUTHORIZATION'] || request.query_parameters[:authorization],
```

Comment:
When are they in the header vs the request_parameters?

Reply:
Unfortunately you can't use headers from webpage javascript.   I was using that from the Ruby code but then just standardized on using the query parameters everywhere.  I can remove the headers option.